src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1126,9 +1130,34 @@ export class EditDataGridPanel extends GridParentComponent {
 		});
 	}
 
-	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): void {
+	onBeforeEditCell(event: Slick.OnBeforeEditCellEventArgs<any>): boolean {
+		let result = true;
 		this.logService.debug('onBeforeEditCell called with grid: ' + event.grid + ' row: ' + event.row
 			+ ' cell: ' + event.cell + ' item: ' + event.item + ' column: ' + event.column);
+
+		let itemToEdit = event.item[event.cell].displayValue;
+
+		if (Services.DBCellValue.isDBCellValue(itemToEdit)) {
```

Comment:
I am not sure why do we have so many different cases to handle, but in line above, the `itemToEdit` is set to `event.item[event.cell].displayValue` and the next line you are calling `itemToEdit.displayValue`, which does seem to be correct, so only branch will actually be used is `typeof itemToEdit === string`.

please simplify the code if the other scenarios are not needed, otherwise, please test the other scenarios.

Reply:
I don't think I am following. please make sure all the code path are tested, especially this PR needs to be ported to release branch.

Reply:
What I meant to say is, after editing a cell, it would replace the DBcell value with a text value until you refresh the grid. 

If you were to edit the same cell again, it would be in the form of a string instead of a DBCell. So yes, the different paths are needed. 

Just tested them, so far, all paths work fine, although the .text one is not hit on my table. (This is meant to be consistent with the column formatter which handles this as well). It should work if the cell contains that value.

Reply:
I assume you have some local changes not pushed, I am still seeing the displayValue being called twice.

Reply:
That is correct Alan.